paddle/fluid/operators/hierarchical_sigmoid_op.h

```diff
@@ -28,20 +30,46 @@ template <typename T, int MajorType = Eigen::RowMajor,
 using EigenMatrix = framework::EigenMatrix<T, MajorType, IndexType>;
 using platform::Transform;
 
+std::vector<int64_t> cal_rows(const framework::LoDTensor& path) {
+  std::set<int64_t> tmp;
+  std::vector<int64_t> rows;
+  rows.clear();
+  for (size_t i = 0; i < static_cast<size_t>(path.dims()[0]); i++) {
+    for (size_t j = 0; j < static_cast<size_t>(path.dims()[1]); j++) {
+      int64_t temp =
+          path.data<int64_t>()[i * static_cast<size_t>(path.dims()[1]) + j];
+      if (temp >= 0) {
+        tmp.insert(temp);
+      }
+    }
+  }
+  rows.assign(tmp.begin(), tmp.end());
+  return rows;
+}
+
 template <typename DeviceContext, typename T>
 class HierarchicalSigmoidOpKernel : public framework::OpKernel<T> {
  public:
   void Compute(const framework::ExecutionContext& ctx) const override {
-    auto* in = ctx.Input<framework::Tensor>("X");
-    auto* w = ctx.Input<framework::Tensor>("W");
-    auto* label = ctx.Input<framework::Tensor>("Label");
-    auto* bias = ctx.Input<framework::Tensor>("Bias");
-    auto* out = ctx.Output<framework::Tensor>("Out");
-    auto* pre_out = ctx.Output<framework::Tensor>("PreOut");
+    auto* in = ctx.Input<framework::LoDTensor>("X");
```

Comment:
What if `X` is null?

Reply:
checked, and change no_null input usage with safe_ref